File: frontend/apps/customer/src/services/patient.ts
import request from './request';

// 就诊人相关接口，主要服务于处方药下单场景。
export interface Patient {
  id: number;
  userId: number;
  name: string;
  idCard: string;
  idCardFront?: string;
  idCardBack?: string;
  phone: string;
  gender: number;
  birthday: string;
  isDefault: number;
}

export interface PatientAddDTO {
  name: string;
  idCard: string;
  idCardFront?: string;
  idCardBack?: string;
  phone: string;
  gender?: number;
  birthday?: string;
  isDefault?: number;
}

export interface PatientUpdateDTO extends PatientAddDTO {
  id: number;
}

// 获取当前用户的就诊人列表。
export const getPatientList = () => {
  return request.get<Patient[]>('/user/patient/list');
};

// 新增就诊人。
export const addPatient = (data: PatientAddDTO) => {
  return request.post('/user/patient/add', data);
};

// 更新就诊人信息。
export const updatePatient = (data: PatientUpdateDTO) => {
  return request.put('/user/patient/update', data);
};

// 删除指定就诊人。
export const deletePatient = (id: number) => {
  return request.delete(`/user/patient/delete/${id}`);
};

File: backend/zhijian-user/src/main/java/com/zhijian/domain/user/entity/Patient.java
package com.zhijian.domain.user.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 就诊人实体
 *
 * @author Liuhaonan
 * @since 1.0.0
 */
@Data
@TableName("sys_patient")
@Schema(description = "就诊人信息")
public class Patient {

    @Schema(description = "主键ID")
    @TableId(type = IdType.AUTO)
    private Long id;

    @Schema(description = "用户ID")
    private Long userId;

    @Schema(description = "姓名")
    @TableField("real_name")
    private String name;

    @Schema(description = "身份证号")
    private String idCard;

    @Schema(description = "身份证正面URL")
    private String idCardFront;

    @Schema(description = "身份证背面URL")
    private String idCardBack;

    @Schema(description = "手机号")
    private String phone;

    @Schema(description = "性别(0-未知 1-男 2-女)")
    private Integer gender;

    @Schema(description = "出生日期")
    private LocalDate birthday;

    @Schema(description = "是否默认(0-否 1-是)")
    private Integer isDefault;

    @Schema(description = "创建时间")
    private LocalDateTime createTime;

    @Schema(description = "更新时间")
    private LocalDateTime updateTime;
}
